File: ColorImage.cpp
#include "opencv2/core/core.hpp"
#include "opencv2/highgui/highgui.hpp"
#include "opencv2/imgproc/imgproc.hpp"
#include "iostream"

 // Author Rishab Shah

using namespace cv;
using namespace std;
 
int main(int argc, char** argv )
{
    Mat src;
    src = imread(argv[1], CV_LOAD_IMAGE_COLOR);
    namedWindow( "Original image", CV_WINDOW_AUTOSIZE );
    imshow( "Original image", src);

	vector<Mat> input_planes(3);
	split(src,input_planes);
	Mat channel1_display, channel2_display, channel3_display;
        imshow("Red",   input_planes[2]);
        imshow("Green",   input_planes[1]);
        imshow("Blue",   input_planes[0]);


	Mat ycrcb_image;
	cvtColor(src, ycrcb_image, CV_BGR2YCrCb);
	split(ycrcb_image,input_planes);
        imshow("Y",   input_planes[0]);
        imshow("Cb",   input_planes[1]);
        imshow("Cr",   input_planes[2]);


	Mat hsv_image;
	cvtColor(src, hsv_image, CV_BGR2HSV);
	vector<Mat> hsv_planes(3);
	split(hsv_image,hsv_planes);
        imshow("Hue",   hsv_planes[0]);
        imshow("Saturation",   hsv_planes[1]);
        imshow("Value",   hsv_planes[2]);


	
    waitKey(0);                                       
    return 0;
} 

File: ColorImage.py
import cv2
image = cv2.imread('Test_images/Lenna.png')


cv2.imwrite("Original Image"+".jpg",image)
input_planes1=cv2.split(image)
cv2.imwrite("RED"+".jpg", input_planes1[2])
cv2.imwrite("Green"+".jpg", input_planes1[1])
cv2.imwrite("Blue"+".jpg", input_planes1[0])

ycrcb_image=cv2.cvtColor(image,cv2.COLOR_BGR2YCrCb)
input_planes2=cv2.split(ycrcb_image)
cv2.imwrite("Y"+".jpg", input_planes2[0])
cv2.imwrite("Cb"+".jpg", input_planes2[1])
cv2.imwrite("Cr"+".jpg", input_planes2[2])

hsv_image=cv2.cvtColor(image,cv2.COLOR_BGR2HSV)
input_planes3=cv2.split(hsv_image)
cv2.imwrite("Hue"+".jpg", input_planes3[0])
cv2.imwrite("Saturation"+".jpg", input_planes3[1])
cv2.imwrite("Value"+".jpg", input_planes3[2])

a=image[20][25]
print('R:',a[1],'G:',a[0],'B:',a[2])

b=ycrcb_image[20][25]
print('Y:',b[0],'Cr:',b[1],'Cb:',b[2])

c=hsv_image[20][25]
print('H:',c[0],'S:',c[1],'V:',c[2])

cv2.waitKey(0)



